bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java

```diff
@@ -104,13 +105,70 @@ public void initialize() {
         if (job == null || job.isCancelled()) {
             pollingJob = scheduler.scheduleWithFixedDelay(this::getStatus, 0, config.refreshInterval, TimeUnit.MINUTES);
         }
     }
 
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case RenaultBindingConstants.CHANNEL_HVAC_TARGET_TEMPERATURE:
+                if (command instanceof RefreshType && !car.isDisableHvac()) {
+                    updateState(CHANNEL_HVAC_TARGET_TEMPERATURE, new QuantityType<Temperature>(
+                            car.getHvacTargetTemperature().doubleValue(), SIUnits.CELSIUS));
+                } else if (command instanceof QuantityType) {
+                    car.setHvacTargetTemperature(((QuantityType<?>) command).doubleValue());
+                    updateState(CHANNEL_HVAC_TARGET_TEMPERATURE, new QuantityType<Temperature>(
+                            car.getHvacTargetTemperature().doubleValue(), SIUnits.CELSIUS));
+                }
+                break;
+            case RenaultBindingConstants.CHANNEL_HVAC_STATUS:
+                // We can only trigger pre-conditioning of the car.
+                if (command instanceof StringType && command.toString().equals(Car.HVAC_STATUS_ON)
+                        && !car.isDisableHvac()) {
+                    final MyRenaultHttpSession httpSession = new MyRenaultHttpSession(this.config, httpClient);
+                    try {
+                        updateState(CHANNEL_HVAC_STATUS, new StringType("PENDING"));
+                        car.resetHVACStatus();
+                        httpSession.initSesssion(car);
+                        httpSession.actionHvacOn(car.getHvacTargetTemperature());
+                        scheduler.schedule(() -> {
+                            updateHvacStatus(httpSession);
+                        }, config.updateDelay, TimeUnit.SECONDS);
+                    } catch (Exception e) {
+                        logger.warn("Error My Renault Http Session.", e);
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    }
+                }
+                break;
+            case RenaultBindingConstants.CHANNEL_CHARGING_MODE:
+                if (command instanceof StringType) {
+                    car.setChargeMode(command.toString());
+                    String mode = car.getChargingMode();
+                    if (mode != null) {
+                        MyRenaultHttpSession httpSession = new MyRenaultHttpSession(this.config, httpClient);
+                        try {
+                            httpSession.initSesssion(car);
+                            httpSession.actionChargeMode(command.toString());
+                            updateState(CHANNEL_CHARGING_MODE, new StringType(mode));
+                        } catch (Exception e) {
+                            httpSession = null;
```

Comment:
Note that it is not done in the other case (channel CHANNEL_HVAC_STATUS).
I let you think if this is normal or not.

Reply:
I need to think about if I can reuse this httpSession for the command update status update.  It is valid for a few minutes (15 from memory) so reusing it might save the re-authentication flow which is quite expensive.  I would need a timeout flag to check and reuse or destroy and recreate...

Reply:
I investigated the default session token and it is valid for only 5 mins.  So I will not try to reuse the session and discard it after each use.

Reply:
Removed the httpSession = null